Mathlib/Tactic/ToAdditive/Frontend.lean

```diff
@@ -699,12 +698,13 @@ def reorderForall (reorder : List (List Nat) := []) (src : Expr) : MetaM Expr :=
 /-- Reorder lambda-binders. See doc of `reorderAttr` for the interpretation of the argument -/
 def reorderLambda (reorder : List (List Nat) := []) (src : Expr) : MetaM Expr := do
   if let some maxReorder := reorder.flatten.max? then
-    lambdaBoundedTelescope src (maxReorder + 1) fun xs e => do
-      if xs.size = maxReorder + 1 then
+    let maxReorder := maxReorder + 1
+    lambdaBoundedTelescope src maxReorder fun xs e => do
+      if xs.size = maxReorder then
         mkLambdaFVars (xs.permute! reorder) e
       else
-        throwError "the permutation\n{reorder}\nprovided by the reorder config option is too \
-          large, the function{indentExpr src}\nhas only {xs.size} arguments"
+        forallBoundedTelescope (← inferType e) (maxReorder - xs.size) fun ys _ => do
```

Comment:
```suggestion
        -- we don't have to consider the case where arguments are too large, since `reorderForall` applied to the type would have already failed in that case.
        forallBoundedTelescope (← inferType e) (maxReorder - xs.size) fun ys _ => do
```
It took me a while to realize why this wouldn't raise panics, so please add a comment. (not sure if this fits in the line limit).